streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -78,14 +78,24 @@
 
     private final Serde<K> keySerde;
     private final Serde<V> valSerde;
+    private final String storeName;
 
     private boolean sendOldValues = false;
 
+
     public KTableImpl(KStreamBuilder topology,
                       String name,
                       ProcessorSupplier<?, ?> processorSupplier,
                       Set<String> sourceNodes) {
-        this(topology, name, processorSupplier, sourceNodes, null, null);
+        this(topology, name, processorSupplier, sourceNodes, null, null, null);
+    }
+
+    public KTableImpl(KStreamBuilder topology,
```

Comment:
It seems incorrect to me that the constructor with store name specified is never used. More specifically:

1. All non-stateful operators should pass along the store name to the new `KTableImpl` object.
2. When `materialize` is called on `KTableSource`, we should set its state store name.
3. For aggregation operators, the resulted `KTableImpl` should set its state store name.
4. For KTable-KTable join operators, the resulted `KTableImpl` should always set state store as null.